src/components/UI/Dropdown/index.js

```diff
@@ -43,9 +50,18 @@ export function StyledDropdown(props) {
           {props?.list.length > 0 ? (
             <DropDownList>
               {props?.list.map((item) => (
-                <ListItem onClick={() => handleSelect(item)}>
-                  {item.name}
-                </ListItem>
+                <React.Fragment>
+                  {(selectedOption?.id !== item.id ||
+                    selectedOption?.name !== item.name) && (
+                    <ListItem onClick={() => handleSelect(item)}>
+                      {item.icon ? (
+                        <Icon src={item.icon ? item.icon : undefined}></Icon>
```

Comment:
same as above, we can use `item.icon` directly

Reply:
since this is the same code as above, it can be abstracted to a method and reused
```
const  getIcon = item => {
  return item.icon ? (
    <Icon src={item.icon ? item.icon : undefined}></Icon>
  ) : (
    item.name
  )}
}
```